### PR TITLE
chore: update all stubs to esp-flasher-stub v0.5.1

### PR DESCRIPTION
**Pull request by jgangemi**
## Summary

- replace all 9 vendored stubs with upstream esp-flasher-stub v0.5.1 release artifacts
- add `tools/update-stubs.sh` for automated stub downloads via `go generate`
- add development docs to README for the go:generate workflow
- fix golangci-lint warnings

## Test plan

- [x] `go test -v ./pkg/espflasher/...` — all stubs load correctly, all tests pass
- [x] `golangci-lint run ./...` — zero issues

### CODE REVIEW

**Comment by deadprogram** [comment]
@jgangemi according to https://github.com/espressif/esp-flasher-stub/blob/master/README.md:

> This project is experimental and not yet ready for production use.

I just wonder if we should wait for the release? Alternately we could switch over entirely, which would mean that this PR should include the updated JSON for all of the currently supported processors.

What do you think?

**Comment by jgangemi** [comment]
i'm going to defer to you on that. i don't have other hardware to test with. i can say this (and the other PRs, some of which were merged) work pretty reliably on the fork branch i've been pointing my projects at, although there still may be a few edge cases.

**Comment by deadprogram** [comment]
Well, I would rather have a PR that switched all of them, if you please. I have several different processors/boards to test with.

**Comment by jgangemi** [comment]
ok - i'll stick claude on it. 😺

**Comment by jgangemi** [comment]
done

File: pkg/espflasher/stub.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"fmt"
 )
 
+//go:generate bash ../../tools/update-stubs.sh
```
Comment (deadprogram) [inline pkg/espflasher/stub.go:10]: We need some docs somewhere on how/when to run `go generate` probably.

**Comment by deadprogram** [comment]
@jgangemi also please remove Claude as co-comitter, as per my previous comment. Thanks!

**Comment by deadprogram** [comment]
@jgangemi can you please rebase this and resolve merge conflict? Thanks!

**Comment by deadprogram** [comment]
Thanks for the update @jgangemi now merging.
